Remove RoomWorkaround (#601)
gradle/android-cache-fix-gradle-plugin: README.md

Fixed by the Android Cache Fix plugin, but unresolved in any current or upcoming preview release of the Android Gradle Plugin:

> * Room annotation processor causes cache misses, doesn't declare outputs, overlapping outputs, etc: https://issuetracker.google.com/issues/132245929

Reviewer: Can we keep a reference to this issue in the readme and also explain that if they encounter this issue, they should use the Room Gradle plugin?

Author: added some context [3eff77c](https://github.com/gradle/android-cache-fix-gradle-plugin/pull/601/commits/3eff77c2fdf708736f006a1690697f2c52b858ba) 
wdyt?

* MergeResources is not relocatable: https://issuetracker.google.com/issues/246529491

### Room

> In the past, the Room annotation processor caused cache misses: https://issuetracker.google.com/issues/132245929.

Reviewer: ```suggestion
The Room annotation processor causes cache misses: https://issuetracker.google.com/issues/132245929.
To work around this issue, please apply the [Room Gradle Plugin](https://developer.android.com/jetpack/androidx/releases/room#2.6.0-alpha02).
```

> We provided the `RoomWorkaround` solving the overlapping outputs and undeclared outputs.

Reviewer: ```suggestion
```

> Since the release of Room Gradle Plugin([2.6.0](https://developer.android.com/jetpack/androidx/releases/room#2.6.0-alpha02)),

Reviewer: ```suggestion
```

> the workaround is no longer required.

Reviewer: ```suggestion
```
